src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -628,7 +700,38 @@ static void copy(final UnixPath source,
 
     // -- native methods --
 
-    static native void transfer(int dst, int src, long addressToPollForCancel)
+    /**
+     * Copies data between file descriptors {@code src} and {@code dst} using
+     * a platform-specific function or system call possibly having kernel
+     * support.
+     *
+     * @param dst destination file descriptor
+     * @param src source file descriptor
+     * @param addressToPollForCancel address to check for cancellation
+     *        (a non-zero value written to this address indicates cancel)
+     *
+     * @return 0 on success, UNAVAILABLE if the platform function would block,
+     *         UNSUPPORTED_CASE if the call does not work with the given
+     *         parameters, or UNSUPPORTED if direct copying is not supported
+     *         on this platform
+     */
+    private static native int directCopy0(int dst, int src,
+                                          long addressToPollForCancel)
+        throws UnixException;
+
+    /**
+     * Copies data between file descriptors {@code src} and {@code dst} using
+     * an intermediate temporary direct buffer.
+     *
+     * @param dst destination file descriptor
+     * @param src source file descriptor
+     * @param address the address of the temporary direct buffer's array
+     * @param size the size of the temporary direct buffer's array
+     * @param addressToPollForCancel address to check for cancellation
+     *        (a non-zero value written to this address indicates cancel)
+     */
+    private static native void bufferCopy0(int dst, int src, long address,
```

Comment:
It might be clearer if the name were bufferedCopy rather than bufferCopy.

Reply:
So changed in 6c7ac10c4a98e0d11cb487dfe816f460f7381002.